framework/test/unit/modules/dpos_v2/method.spec.ts

```diff
@@ -140,4 +147,190 @@ describe('DposModuleApi', () => {
+				{ tokenID: tokenID2, coefficient: Buffer.alloc(MAX_NUMBER_BYTES_Q96) },
+				{ tokenID: tokenID3, coefficient: Buffer.alloc(MAX_NUMBER_BYTES_Q96) },
+			],
+		};
+		const delegateData2 = {
+			consecutiveMissedBlocks: 0,
+			isBanned: false,
+			lastGeneratedHeight: 5,
+			name: 'delegate2',
+			pomHeights: [],
+			selfVotes: BigInt(0),
+			totalVotesReceived: BigInt(4),
+			commission: 0,
+			lastCommissionIncreaseHeight: 0,
+			sharingCoefficients: [
+				{ tokenID: tokenID1, coefficient: Buffer.alloc(MAX_NUMBER_BYTES_Q96) },
+				{ tokenID: tokenID2, coefficient: Buffer.alloc(MAX_NUMBER_BYTES_Q96) },
+				{ tokenID: tokenID3, coefficient: Buffer.alloc(MAX_NUMBER_BYTES_Q96) },
+			],
+		};
+		const delegateData3 = {
+			consecutiveMissedBlocks: 0,
+			isBanned: false,
+			lastGeneratedHeight: 5,
+			name: 'delegate3',
+			pomHeights: [],
+			selfVotes: BigInt(0),
+			totalVotesReceived: BigInt(4),
+			commission: 0,
+			lastCommissionIncreaseHeight: 0,
+			sharingCoefficients: [
+				{ tokenID: tokenID2, coefficient: Buffer.from('a40000000000000000000000', 'hex') },
+				{ tokenID: tokenID1, coefficient: Buffer.alloc(MAX_NUMBER_BYTES_Q96) },
+				{ tokenID: tokenID3, coefficient: Buffer.alloc(MAX_NUMBER_BYTES_Q96) },
+			],
+		};
+		const defaultConfig = {
+			factorSelfVotes: 10,
+			maxLengthName: 20,
+			maxNumberSentVotes: 10,
+			maxNumberPendingUnlocks: 20,
+			failSafeMissedBlocks: 50,
+			failSafeInactiveWindow: 260000,
+			punishmentWindow: 780000,
+			roundLength: 103,
+			minWeightStandby: BigInt(1000) * BigInt(10 ** 8),
+			numberActiveDelegates: 101,
+			numberStandbyDelegates: 2,
+			governanceTokenID: Buffer.from('0000000000000000', 'hex'),
+			tokenIDFee: Buffer.from('0000000000000000', 'hex'),
+			delegateRegistrationFee: BigInt(10) * BigInt(10) ** BigInt(8),
+			maxBFTWeightCap: 500,
+			commissionIncreasePeriod: COMMISSION_INCREASE_PERIOD,
+			maxCommissionIncreaseRate: MAX_COMMISSION_INCREASE_RATE,
+		};
+		let tokenMethod: any;
+
+		beforeEach(async () => {
+			tokenMethod = { lock: jest.fn() };
+			dposMethod.init(moduleName, defaultConfig, tokenMethod);
+			await delegateSubStore.set(createStoreGetter(stateStore), address1, delegateData1);
+			await delegateSubStore.set(createStoreGetter(stateStore), address2, delegateData2);
+			await delegateSubStore.set(createStoreGetter(stateStore), address3, delegateData3);
+			jest.spyOn(tokenMethod, 'lock');
+			jest.spyOn(delegateSubStore, 'set');
+		});
+
+		it('should return if totalVotesReceived is 0', async () => {
+			await expect(
+				dposMethod.updateSharedRewards(methodContext, address1, tokenID1, BigInt(50)),
+			).resolves.toBeUndefined();
+			expect(tokenMethod.lock).not.toHaveBeenCalled();
+			expect(delegateSubStore.set).not.toHaveBeenCalled();
+		});
+
+		it('should initialize sharing coefficient to zero and set the appropriate amounts to delegate store for the specified token if there does not exist an item in delegateStore for the token id', async () => {
```

Comment:
here the new tokenID should be sorted (maybe it's better to check the case where sharing coefficient is added in hte middle of the list

Reply:
new tokenID is sorted correctly in the output and checked with the expectation. The case where sorting is checked for sharing coefficient in the middle of the list is also checked in the next/last test case. I think they cover all the cases for sorting.